### PR TITLE
build: publish@4.2.0-beta.1

### PR DESCRIPTION
**Pull request by ZLY201**
Publish version 4.2.0-beta.1.

Expected to be released at 11.06.

**note: The last commit must be build: publish@{version}.**

### CODE REVIEW

**Comment by ZLY201** [comment]
@noname0310 help to review

File: package.json [hunk 1]
```diff
@@ -68,10 +73,11 @@
     "commitlint": "^17.0.3",
     "compare-versions": "^5.0.1",
     "conventional-changelog-conventionalcommits": "^5.0.0",
-    "crypto": "^1.0.1",
```
Comment (noname0310) [inline package.json:71]: Why was the `crypto` removed? Is there no problem with the build?

File: tools/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (noname0310) [inline tools/tsconfig.json:1]: What is this tsConfig file used for?

File: README.md [hunk 3]
```diff
@@ -224,3 +224,73 @@ Thanks also give to these sponsors or backers:
...
+[stack-npm-link]: https://www.npmjs.com/package/@js-sdsl/stack
+[stack-umd-size]: https://img.badgesize.io/https://unpkg.com/@js-sdsl/stack/dist/umd/stack.min.js?compression=gzip&style=flat-square/
+[stack-umd-link]: https://unpkg.com/@js-sdsl/stack/dist/umd/stack.min.js
+[stack-docs]: https://js-sdsl.org/js-sdsl/classes/Stack.html
+
+[queue-package]: https://github.com/js-sdsl/js-sdsl/blob/main/src/container/OtherContainer/Queue.ts
+[queue-npm-version]: https://img.shields.io/npm/v/@js-sdsl/queue
+[queue-npm-link]: https://www.npmjs.com/package/@js-sdsl/queue
+[queue-umd-size]: https://img.badgesize.io/https://unpkg.com/@js-sdsl/queue/dist/umd/queue.min.js?compression=gzip&style=flat-square/
+[queue-umd-link]: https://unpkg.com/@js-sdsl/queue/dist/umd/stack.min.js
```
Comment (noname0310) [inline README.md:239]: fix to:
[queue-umd-link]: https://unpkg.com/@js-sdsl/queue/dist/umd/queue.min.js

File: package.json (hunk 1 above)
Comment (ZLY201) [reply package.json:71]: `crypto` is a inline module in `node`. The package is be deprecated.

File: tools/tsconfig.json (hunk 2 above)
Comment (ZLY201) [reply tools/tsconfig.json:1]: For the files in `tools` the enviroment should be `node` but not `browser`.

There will be some wrong message in my `webstorm` without this `tsconfig`.

File: README.md (hunk 3 above)
Comment (ZLY201) [reply README.md]: done

**Review by noname0310 — APPROVED** [review]

**Pull request merged**